commons/src/main/java/io/aiven/kafka/tieredstorage/commons/ObjectKey.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.commons;
+
+import java.text.NumberFormat;
+import java.util.Objects;
+
+import org.apache.kafka.common.TopicIdPartition;
+import org.apache.kafka.server.log.remote.storage.RemoteLogSegmentId;
+import org.apache.kafka.server.log.remote.storage.RemoteLogSegmentMetadata;
+
+/**
+ * Maps Kafka segment files to object paths/keys in the storage backend.
+ */
+public final class ObjectKey {
+    /**
+     * Supported files and extensions, including log, index types, and segment manifest.
+     *
+     * @see org.apache.kafka.server.log.remote.storage.RemoteStorageManager.IndexType
+     */
+    public enum Suffix {
```

Comment:
Do we really need this enum? AFAIU we do not use the actual `enum` functionality and to map this suffixes to actual index types we will need to do some mapping so why not just use Kafka's `IndexType` enum like we were doing before and constants for manifest and log files?

Reply:
Agree this is missing. I guess we could have a static method within the enum to do the mapping? 
```java
  public enum Suffix {
        // ...
        static Suffix from(RemoteStorageManager.IndexType indexType) {
            switch (indexType) {
                case LEADER_EPOCH: return LEADER_EPOCH_CHECKPOINT;
                case TRANSACTION: return TXN_INDEX;
                case OFFSET: return OFFSET_INDEX;
                case TIMESTAMP: return TIME_INDEX;
                case PRODUCER_SNAPSHOT: return PRODUCER_SNAPSHOT;
                default: throw new IllegalArgumentException("Unknown index type " + indexType);
            }
        }
  }
```
So, when copying files we could pass Suffix.from(...) when we know the index type, and Suffix.LOG/MANIFEST for the other cases.

Reply:
Well, this makes sense. I added this static conversion